views/charts.tmpl

```diff
@@ -5,69 +5,142 @@
+                                <a
+                                    class="nav-link"
+                                    href="javascript:void(0);"
+                                    data-target="charts.zoomOption"
+                                    data-action="click->charts#onZoom"
+                                    data-zoom="8760"
+                                >Month</a>
+                            </li>
+                            <li class="nav-item">
+                                <a
+                                    class="nav-link"
+                                    href="javascript:void(0);"
+                                    data-target="charts.zoomOption"
+                                    data-action="click->charts#onZoom"
+                                    data-zoom="2016"
+                                >Week</a>
+                            </li>
+                            <li class="nav-item">
+                                <a
+                                    class="nav-link"
+                                    href="javascript:void(0);"
+                                    data-target="charts.zoomOption"
+                                    data-action="click->charts#onZoom"
+                                    data-zoom="288"
+                                >Day</a>
+                            </li>
+                        </ul>
                     </div>
                 </div>
-     
+
+                <div class="chart-control-wrapper mb-1" data-target="charts.zoomSelector">
+                    <div class="chart-control-label">MOVING AVG</div>
+                    <div class="chart-control">
+                        <input
+                            type="number"
+                            min="1"
+                            max="10000"
+                            class="form-control"
+                            style="max-width: 80px"
+                            placeholder="&mdash;"
+                            data-target="charts.rollPeriodInput"
+                            data-action="change->charts#onRollPeriodChange"
+                        >
+                    </div>
+                </div>
+
             </div>
-            <br><br>
-            <div id="graphdiv" 
-                data-target="chartsview" 
-                style="width:100%; height:450px; margin:0 auto;">
+
+
+        </div>
+
+
+        <div data-target="charts.chartWrapper" class="chart-wrapper pl-2 pr-2">
+            <div
+                data-target="charts.chartsView"
+                style="width:100%; height:76vh; margin:0 auto;">
+            </div>
+            <div class="spinner-wrapper">
+                <div class="spinner-centerer d-flex align-items-center justify-content-center">
+                    <div class="spinner">
+                        <i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i>
```

Comment:
What do all these `<i>`s do?

Reply:
They are the spokes of the spinner wheel
<img width="637" alt="spinner" src="https://user-images.githubusercontent.com/25571523/44196915-1d5d8c80-a0f2-11e8-9376-6ccdce3ba928.png">